mono/mini/interp/interp.c

```diff
@@ -3441,28 +3441,30 @@ interp_exec_method_full (InterpFrame *frame, ThreadContext *context, FrameClause
 				MONO_PROFILER_RAISE (method_tail_call, (frame->imethod->method, new_method->method));
 
 			if (!new_method->transformed) {
-				MONO_API_ERROR_INIT (error);
+				error_init_reuse (error);
 
 				frame->ip = ip;
 				mono_interp_transform_method (new_method, context, error);
 				MonoException *ex = mono_error_convert_to_exception (error);
 				if (ex)
 					THROW_EX (ex, ip);
 			}
-			ip += 2;
 			const gboolean realloc_frame = new_method->alloca_size > frame->imethod->alloca_size;
-			frame->imethod = new_method;
 			/*
 			 * We allocate the stack frame from scratch and store the arguments in the
 			 * locals again since it's possible for the caller stack frame to be smaller
 			 * than the callee stack frame (at the interp level)
 			 */
 			if (realloc_frame) {
-				frame->stack = (stackval*)g_alloca (frame->imethod->alloca_size);
-				memset (frame->stack, 0, frame->imethod->alloca_size);
+				if (frame->imethod->has_localloc)
+					frame->stack = (stackval*)g_alloca (new_method->alloca_size);
+				else
+					frame->stack = (stackval*)g_alloca (new_method->alloca_size - frame->imethod->alloca_size);
```

Comment:
I can't say I understand much from this PR. Here we set frame->stack with allocated memory on the stack that is not enough (shouldn't frame->stack always point to at least new_method->alloca_size amount of memory). What if new_method->alloca_size < frame->imethod->alloca_size. This PR cares about LOCALLOC but not about other sources of alloca ?

Reply:
I think there are just two sources: `localloc` and `alloca_size`.

The point is that `alloca` is adjacent/additive.
You can reuse the `alloca` of the conceptual caller towards production of the alloca for the callee.
If the caller has 10 bytes and the callee needs 30, you can get by with just 20 more.

Unless there is another allocation in between the localloc -- which might have other problems, but ignored here.

I think those are the only two allocas. If I messed some, then that is a mistake.

Reply:
> Unless there is another allocation in between the localloc -- which might have other problems, but ignored here.

That sounds very easy to break for little benefit.

Reply:
Jmp isn’t supposed to keep growing the stack.
To really implement this correctly,
 need to return to caller and then do the full alloca
there and then call back to here.
Kinda like the “split” but only for jmp.

Reply:
You haven't answered my first 2 questions